### ISSUE TITLE
Support for API Platform

### ISSUE THREAD

**Issue by vincentchalamon** [dunglas/doctrine-json-odm#6]
Hi,

I'm trying to implement this tool on my Symfony 3 project using API Platform 2. Here is my schema:
```php
/**
 * @ORM\Entity
 *
 * @Resource(attributes={
 *     "normalization_context"={"groups"={"dues_default_output"}},
 *     "denormalization_context"={"groups"={"dues_default_input"}}
 * })
 */
class Dues
{
    /**
     * @var int
     *
     * @ORM\Id
     * @ORM\Column(type="integer")
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    /**
     * @var Club
     *
     * @ORM\Column(type="json_document", options={"jsonb": true})
     *
     * @Groups({"dues_default_output", "dues_default_input"})
     */
    private $club;

    /**
     * @var bool
     *
     * @ORM\Column(type="string")
     *
     * @Groups({"dues_default_output", "dues_default_input"})
     */
    private $foo;
}
```

I'm trying to create a `Dues` object with its `Club` object. Here is the request:
```json
{
    "foo": "bar",
    "club": {
        "name": "US CAGNES TENNIS",
        "website": "http://www.uscagnestennis.org",
        "address": "Avenue du colonel Jean-Pierre",
        "zipcode": "06800",
        "city": "Cagnes sur mer"
    }
}
```

Maybe API Platform 2 doesn't support this yet, or maybe it's not related to API Platform neither, but I get this error:
```
No resource class found for object of type "LesTilleuls\\ClubBundle\\Entity\\Club"
```

Any idea how could I fix it?

**Comment by vincentchalamon** [comment]
I think I can eventually convert my `club` object through `Club` PHP object on `kernel.request` event. I'll try & comment here

**Comment by vincentchalamon** [comment]
Does not work neither…

**Comment by vincentchalamon** [comment]
My bad, seen with Hamza: request worked well, I just needed to create a custom normalizer to manage this field. But it can be wonderful to have it automatically in API Platform, what do you think @dunglas?

**Comment by dunglas** [comment]
What do you store in the `$club` property? API Platform should be able to serialize and deserialize objects without custom normalizer (it will to default to the built-in Symfony `ObjectNormalizer`). Array of objects yet because of a limitation of the Symfony Serializer.

Anyway, it looks like an API Platform bug, not a JSON Doctrine ODM one.

**Comment by vincentchalamon** [comment]
Well, I first developed my own ClubNormalizer cause API Platform wasn't able to retrieve club id (looks like it expected Club to be a valid resource, but it was just a Club object: not a Doctrine entity, not an API Platform resource, just an object 😉).
I've updated my dependencies, and I don't need my custom normalizer anymore. API Platform update? Other dependency update? Anyway, it works now
